v3-roadmap-and-design.md

```diff
@@ -0,0 +1,425 @@
+# Zarr Python Roadmap
```

Comment:
This is formatted w/ markdown. If folks think this doc would fit better as a page in the docs, we can convert it to RST. cc @MSanKeys963